cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java

```diff
@@ -246,16 +246,16 @@ private void restartIfNecessary(int podId, boolean finalAttempt)
             try {
                 try {
                     adminClient = adminClient(podId);
-                    Integer controller = controller(podId, adminClient, 1, TimeUnit.MINUTES);
+                    Integer controller = controller(podId, adminClient, operationTimeoutMs, TimeUnit.MILLISECONDS);
                     int stillRunning = podToContext.reduceValuesToInt(100, v -> v.future.isComplete() ? 0 : 1,
                             0, Integer::sum);
                     if (controller == podId && stillRunning > 1) {
                         log.debug("Pod {} is controller and there are other pods to roll", podId);
                         throw new ForceableProblem("Pod " + podName(podId) + " is currently the controller and there are other pods still to roll");
                     } else {
-                        if (canRoll(adminClient, podId, 1, TimeUnit.MINUTES)) {
+                        if (canRoll(adminClient, podId, operationTimeoutMs, TimeUnit.MILLISECONDS)) {
```

Comment:
@tombentley What about this one? Thsi used significantly different timeout before and I wonder if it should stay this way.

Reply:
Yeah, the `operationTimeoutMs` should be the `STRIMZI_OPERATION_TIMEOUT_MS` env var from the CO deployment which default to 300000 ms and now we are moving from 1 minute to 5 minutes.

Reply:
Well obviously if we're switching to 300000ms that would appear to make the existing problem worse, not better. 
I think it would be better to make an algorithms change than to tinker with timeouts because if you just reduce the timeouts so it takes a reasonable amount of time in the known problematic case you're making them too short for some other (as yet unknown) problematic case(s).

Reply:
I think these are only slightly related issues. IMHO all of the readiness timeouts etc. should be using the operation timeout. This one is the only one which probably should not without some further changes.

Reply:
@scholzj true.